sample/src/androidTest/java/com/juul/pommel/sample/PommelSampleTest.kt

```diff
@@ -20,7 +20,8 @@ import javax.inject.Named
     value = [
         EnglishWelcomeProvider_SoloModule::class,
         EnglishNameProvider_SoloModule::class,
-        subGreeting_SoloModule::class
+        subGreeting_SoloModule::class,
```

Comment:
Being that it is generating a class, do we want it to conform to class naming conventions? (i.e. uppercase first letter?)

Reply:
@twyatt I took this convention from `Dagger` and `AssistedInject` both libraries add underscores to any generated class to make distinction to the developer that they are calling generated code. We don't have to follow this rule as it up to us how we name this files. We can follow a more normal naming convention to allow for easier discovery for developers

Reply:
IMHO `SubGreeting_SoloModule` is more familiar (as being a class) but I'll leave it up to you.

Reply:
@twyatt I took @cedrickcooke suggestion and functions will generate the following structure for the file name. 

`FileName_FunctionName_SoloModule` 

This is to avoid any collisions in `object` or `companion` classes that have the same function name. Which can be highly likely if you are using `@SoloModule` to generate a factory 

so for the above example 

```kotlin
@file:JvmName("ConfigKt")

@SoloModule
@Named("subGreeting")
fun subGreeting(): String {
    return "Good Morning!"
}
```

would generate 

`ConfigKt_subGreeting_SoloModule` for the file name